### PR TITLE
ci: publish build snapshots from CircleCI, not Travis

### PR DESCRIPTION
**Pull request by alexeagle**
Notes:
- uses the same GitHub auth mechanism as we do on Travis
- we need a symmetric key in the circleCI UI because they don't have a dedicated public/private key pair for encrypted variables

### CODE REVIEW

File: docs/DEVELOPER.md [hunk 1]
```diff
@@ -133,11 +133,9 @@ $ gulp lint
 
 ## Publishing snapshot builds
 
-When the `master` branch successfully builds on Travis, it automatically publishes build artifacts
+When the `master` branch successfully builds on CircleCI, it automatically publishes build artifacts
```
Comment (IgorMinar) [inline docs/DEVELOPER.md]: This applies to any upstream branch not just master. Please update.

**Review by IgorMinar — APPROVED** [review]
The rest looks good

**Comment by gkalpak** [comment]
Travis didn't like this change 😛

**Review by IgorMinar — COMMENTED** [review]
circle ci still fails with `/home/circleci/ng/integration/../scripts/ci/payload-size.sh: line 92: ANGULAR_PAYLOAD_FIREBASE_TOKEN: unbound variable`

one job on travis failed as well, but it might have been just a flake, I restarted the job.

**Comment by alexeagle** [comment]
Currently this PR has two commits on it, but I'm unable to get a value for `TRAVIS_COMMIT_RANGE` that works.

In the SSH debug session on circleCI, I see
```
$ echo $CIRCLE_COMPARE_URL
https://github.com/angular/angular/compare/53184f16c572...43e71ab1f150
```
Viewing that page on GitHub, we see only one commit
9utwxctvkx1

What's more, the left end of that range references a commit that's not in the history of the git clone circleCI pulled down:

```
$ git log --oneline 53184f16c572...43e71ab1f150 --
fatal: bad revision '53184f16c572...43e71ab1f150'
```

**Pull request closed without merging**